tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu

```diff
@@ -211,55 +211,58 @@ RUN ${PIP} --no-cache-dir install --upgrade \
 RUN ln -s $(which ${PYTHON}) /usr/local/bin/python \
    && ln -s $(which ${PIP}) /usr/bin/pip
 
+# Install extra packages
+RUN $PYTHON -m pip install --no-cache-dir \
+    "bokeh>=2.3,<2.4" \
+    "imageio==2.13.3" \
+    "opencv-python>=4.3,<5" \
+    "plotly>=5.1,<6" \
+    "seaborn>=0.11,<1" \
+    "shap>=0.39,<1"
+ 
 # install PyYAML==5.1.2 to avoid conflict with latest awscli
 # # python-dateutil==2.8.0 to satisfy botocore associated with latest awscli
-RUN ${PIP} install --no-cache-dir -U \
+RUN $PYTHON -m pip install --no-cache-dir \
    pybind11 \
    cmake==3.18.2.post1 \
-   numpy==1.19.5 \
    scipy==1.7.0 \
    scikit-learn==0.24.2 \
    pandas==1.2.5 \
-   "Pillow>=8.3,<8.4" \
+   Pillow==9.0.1 \
    python-dateutil==2.8.1 \
    "pyYAML>=5.4,<5.5" \
-   requests==2.25.1 \
+   requests==2.27.1 \
    "awscli<2" \
    mpi4py==3.0.3 \
-   "sagemaker>=2,<3" \
-   sagemaker-experiments==0.* \
-   "sagemaker-tensorflow>=2.6,<2.7" \
-   "sagemaker-tensorflow-training>=20" \
    "shap>=0.39,<1" \
    "boto3>1.0,<2.0" \
    "sasl<1" \
    "thrift_sasl<1" \
    "PyHive<1" \
    "sparkmagic<1" \
+   ${TF_URL} \
+   tensorflow-datasets==4.5.2 
+
+RUN $PYTHON -m pip install --no-cache-dir \
+   "sagemaker==2.70.0" \
```

Comment:
Why pin sagemaker version to 2.70 specifically?

Reply:
`sagemaker` needs attrs==20.3.0, which is behind the latest version installed by `sparkmagic`. Therefore using 2.70 version that does not have this limitation.